composer.json

```
@@ -29,7 +29,7 @@
         "php": ">=7.1",
         "evenement/evenement": "^3.0 || ^2.0 || ^1.0",
         "fig/http-message-util": "^1.1",
-        "psr/http-message": "^1.0",
+        "psr/http-message": "^2.0 || ^1.0",
```

The only way this will work is if we go:

```suggestion
        "psr/http-message": "^2.0",
```

Reason for this is that we have a bunch of messages in https://github.com/reactphp/http/tree/3.x/src/Message that don't have a return type on those methods. If we'll allow 2.0 without adding that you'll get fatal errors, so IMHO we should switch to 2.0 and add the return types to all the relevant classes. /cc @clue

---

@WyriHaximus https://github.com/reactphp/http/pull/541 was not what you expect?

---

@L3o-pold Aside from a few things, yes that should have worked.

---

@WyriHaximus can you clarify what so we can move forward?

---

@valzargaming I have opened a [PR](https://github.com/valzargaming/http/pull/1) against your branch to add the minimum required to make this update work. I had to remove clue/http-proxy-react because it is only required for 1 example, and it eventually requires [ringcentral/psr7](https://github.com/ringcentral/psr7/blob/master/composer.json#L16) which in turn depends on "psr/http-message": "^1.0".  You will need to rebase after merge because your branch is 2 commits behind and I didn't want to introduce a merge commit.

---

@valzargaming sorry, one more thing. I didn't realize the CI ran against 7.1, you will need to remove https://github.com/reactphp/http/blob/3.x/.github/workflows/ci.yml#L22

---

No worries. I don't have access to my PC right now, but if you make a PR for it I'll merge it.

---

@valzargaming PR opened.

---

@L3o-pold As mentioned in https://github.com/clue/reactphp-http-proxy/issues/64#issuecomment-2960449520 it boils down to this function call: https://github.com/clue/reactphp-http-proxy/blob/1.x/src/ProxyConnector.php#L187 Mento to respond to you earlier but life got in the way. Had a look at fixing it but didn't get far